crates/shared/src/http_solver.rs

```diff
@@ -78,14 +78,12 @@ impl HttpSolverApi for DefaultHttpSolverApi {
         let mut url = self.base.clone();
         url.set_path("/solve");
 
-        let instance_name = self.generate_instance_name(
-            model
-                .metadata
-                .as_ref()
-                .unwrap_or(&MetadataModel::default())
-                .auction_id
-                .unwrap_or(0u64),
-        );
+        let maybe_auction_id = model
+            .metadata
+            .as_ref()
+            .unwrap_or(&MetadataModel::default())
+            .auction_id;
+        let instance_name = self.generate_instance_name(maybe_auction_id.unwrap_or(0u64));
```

Comment:
nit: The value of `maybe_auction_id` can depend on the implementation of `MetadataModel::default()` which is probably not intended. As an alternative you could try this:
```rust
        let auction_id = match &model.metadata {
            Some(metadata) => metadata.auction_id.unwrap_or(0),
            None => 0,
        };
```
I believe this makes a little more obvious what's going on.
Alternatively if you actually want this value to be an `Option<u64>`  this would also work:
```rust 
        let maybe_auction_id = model.metadata.as_ref().and_then(|data| data.auction_id);
```

Reply:
I'll go with the last one, thank you!